test/functional/p2p_private_broadcast.py

```diff
@@ -0,0 +1,311 @@
+        self.wait_until(lambda: len(node1.getrawmempool()) > 0)
+        far_observer.wait_for_tx(tx["txid"])
+        self.log.debug(f"Outbound connection i={NUM_INITIAL_CONNECTIONS}: the private broadcast target received and further relayed the transaction")
+
+        num_private_broadcast_opened = 1 # already 1 connection to node1, confirmed by far_observer getting the tx
+        for i in range(NUM_INITIAL_CONNECTIONS + 1, len(self.socks5_server.conf.destinations)):
+            self.log.debug(f"Inspecting outbound connection i={i}")
+            # At this point the connection may not yet have been established (A),
+            # may be active (B), or may have already been closed (C).
+            peer = self.socks5_server.conf.destinations[i]["node"]
+            peer.wait_until(lambda: peer.message_count["version"] == 1, check_connected=False)
+            # Now it is either (B) or (C).
+            requested_to_addr = self.socks5_server.conf.destinations[i]["requested_to_addr"]
+            if peer.last_message["version"].nServices != 0:
+                self.log.debug(f"Outbound connection i={i} to {requested_to_addr} not a private broadcast, ignoring it (maybe feeler or extra block only)")
+                continue
+            self.log.debug(f"Outbound connection i={i} to {requested_to_addr} must be a private broadcast, checking it")
+            assert requested_to_addr.endswith(".onion")
+            peer.wait_for_disconnect()
+            # Now it is (C).
+            assert_equal(peer.message_count, {
+                "version": 1,
+                "verack": 1,
+                "tx": 1,
+                "ping": 1
+            })
+            dummy_address = CAddress()
+            dummy_address.nServices = 0
+            assert_equal(peer.last_message["version"].nVersion, P2P_VERSION)
+            assert_equal(peer.last_message["version"].nServices, 0)
+            assert_equal(peer.last_message["version"].nTime, 0)
+            assert_equal(peer.last_message["version"].addrTo, dummy_address)
+            assert_equal(peer.last_message["version"].addrFrom, dummy_address)
+            assert_equal(peer.last_message["version"].strSubVer, "/pynode:0.0.1/")
+            assert_equal(peer.last_message["version"].nStartingHeight, 0)
+            assert_equal(peer.last_message["version"].relay, 0)
+            assert_equal(peer.last_message["tx"].tx.rehash(), tx["txid"])
+            self.log.debug(f"Outbound connection i={i} is proper private broadcast, test ok")
+            num_private_broadcast_opened += 1
+            if num_private_broadcast_opened == NUM_PRIVATE_BROADCAST_PER_TX:
+                break
+        assert_equal(num_private_broadcast_opened, NUM_PRIVATE_BROADCAST_PER_TX)
+
+        wtxid_int = int(tx["wtxid"], 16)
+        inv = CInv(MSG_WTX, wtxid_int)
+
+        self.log.info("Checking that the transaction is not in the originator node's mempool")
+        assert len(tx_originator.getrawmempool()) == 0
```

Comment:
d10a0649b018638574fa7dc316b3e783bf6cee69 sanity check: you can assert this because none of the private-broadcast peers are connected back to the `tx_originator`?

Reply:
Yes, exactly.